support-frontend/assets/components/directDebit/directDebitProgressiveDisclosure/directDebitForm.jsx

```diff
@@ -0,0 +1,266 @@
+  accountNumber: Object,
+}
+
+
+// ----- Map State/Props ----- //
+
+function mapStateToProps(state) {
+  return {
+    sortCodeString: state.page.directDebit.sortCodeString,
+    accountNumber: state.page.directDebit.accountNumber,
+    accountHolderName: state.page.directDebit.accountHolderName,
+    accountHolderConfirmation: state.page.directDebit.accountHolderConfirmation,
+    formError: state.page.directDebit.formError,
+    countryGroupId: state.common.internationalisation.countryGroupId,
+  };
+}
+
+function mapDispatchToProps(dispatch: Dispatch<Action>) {
+
+  return {
+    payDirectDebitWithoutConfirmation: () => {
+      dispatch(payDirectDebitWithoutConfirmation());
+      return false;
+    },
+    confirmDirectDebitClicked: (onPaymentAuthorisation: PaymentAuthorisation => void) => {
+      dispatch(confirmDirectDebitClicked(onPaymentAuthorisation));
+      return false;
+    },
+    updateSortCodeString: (event: SyntheticInputEvent<HTMLInputElement>) => {
+      dispatch(updateSortCodeString(event.target.value));
+    },
+    updateAccountNumber: (event: SyntheticInputEvent<HTMLInputElement>) => {
+      const accountNumber: string = event.target.value;
+      dispatch(updateAccountNumber(accountNumber));
+    },
+    updateAccountHolderName: (event: SyntheticInputEvent<HTMLInputElement>) => {
+      const accountHolderName: string = event.target.value;
+      dispatch(updateAccountHolderName(accountHolderName));
+    },
+  };
+
+}
+
+const InputWithError = compose(withError, withLabel)(Input);
+
+
+// ----- Component ----- //
+
+class DirectDebitForm extends Component<PropTypes, StateTypes> {
+  constructor(props: PropTypes) {
+    super(props);
+    this.state = {
+      accountHolderName: {
+        error: '',
+        message: 'Please enter a valid account name',
+        // Regex matches a string with any character that is not a digit
+        rule: accountHolderName => accountHolderName.match(/^\D+$/),
+      },
+      sortCodeString: {
+        error: '',
+        message: 'Please enter a valid sort code',
+        // Regex matches a string with exactly 6 digits
+        rule: sortCodeString => sortCodeString.match(/^\d{6}$/),
+      },
+      accountNumber: {
+        error: '',
+        message: 'Please enter a valid account number',
+        // Regex matches a string with between 6 and 10 digits
+        rule: accountNumber => accountNumber.match(/^\d{6,10}$/),
+      },
+    };
+  }
+
+  onSubmit = (event) => {
+    event.preventDefault();
+    const cardErrors = this.validateForm();
+    if (this.props.allErrors.length === 0 && !cardErrors) {
+      this.submitForm();
+    }
+  };
+
+  onChange = (field, onChange, event) => {
+    this.setState({
+      [field]: {
+        ...this.state[field],
+        error: '',
+      },
+    });
+    onChange(event);
+  }
+
+  getCardErrors = () => {
+    const { state } = this;
+    const cardErrors = ['accountHolderName', 'sortCodeString', 'accountNumber'].map(field =>
```

Comment:
could we set these magic strings as constants as they are being used in multiple places e.g. `ACCOUNT_HOLDER_NAME = 'accountHolderName';`

Reply:
Nice! Thanks